src/USER-MESONT/pair_mesont_tpm.cpp

```diff
@@ -510,6 +510,12 @@ void PairMESONTTPM::compute(int eflag, int vflag){
 
   // set per atom values and accumulators
   // reallocate per-atom arrays if necessary
+  if (eatom_s == nullptr)
```

Comment:
This is likely the wrong fix. It should not be needed, if `maxeatom` is properly initialized.

Reply:
@akohlmey , Could you please tell me where the initial allocation of 'eatom' array is made? I will make sure that the variables added in the mesont package: 'eatom_s', 'eatom_b', 'eatom_t' are treated in the same way, because it seems that there is an issue with the initial allocation.

Reply:
eatom is the wrong item to compare to.

the situation is quite clear: `eatom_s`, `eatom_b`, and `eatom_t` are initialized to `nullptr` in the pair style constructor, so the only case where they are not increased by this code segment is if `maxeatom` is larger or equal than `atom->nmax`. but that should not happen unless `maxeatom` is not initialized to 0 in the pair style constructor.

If you need some code to compare to, you could look at the "rho" array in  pair style eam.

BTW: I notice that your code deallocates and allocates `eatom`, that should *not* be done. This array is managed by the base class.

Reply:
as background information: the allocation of eatom is triggered only if there is a "consumer" for it, which results in the eflag_atom (and eflag_either) variable become "1" instead of "0". So if you have problems accessing eatom, you have to prefix that with corresponding tests, so that it is not set when it is not requested and thus not allocated.

Reply:
Thank you so much for your clarification, I'll look more into it.